packages/core/src/manifest.ts

```diff
@@ -58,58 +62,80 @@ async function getDevelopmentInstanceId(provider: EthereumProvider, chainId: num
     );
   }
 
-  return hardhatMetadata.instanceId;
+  return {
+    networkName: 'hardhat',
+    instanceId: hardhatMetadata.instanceId,
+    forkedNetwork: hardhatMetadata.forkedNetwork,
+  };
 }
 
-function getSuffix(chainId: number, devInstanceId?: string) {
-  if (devInstanceId !== undefined) {
-    return `${chainId}-${devInstanceId}`;
+function getSuffix(chainId: number, devInstanceMetadata?: DevInstanceMetadata) {
+  if (devInstanceMetadata !== undefined) {
+    return `${chainId}-${devInstanceMetadata.instanceId}`;
   } else {
     return chainId;
   }
 }
 
-async function makeAndGetTempDir() {
-  const tempDir = path.join(os.tmpdir(), MANIFEST_TEMP_DIR);
-  await fs.mkdir(tempDir, { recursive: true });
-  return tempDir;
+interface DevInstanceMetadata {
+  networkName: string;
+  instanceId: string;
+  forkedNetwork?: {
+    // The chainId of the network that is being forked
+    chainId: number;
+  };
 }
 
 export class Manifest {
   readonly chainId: number;
-  readonly devInstanceId?: string;
   readonly file: string;
-  readonly manifestDir: string;
-  private readonly fallbackFile: string;
+  readonly fallbackFile: string;
+
+  private readonly devInstanceMetadata?: DevInstanceMetadata;
+  readonly devDir?: string;
+  readonly devFile?: string;
+
+  private readonly forked: boolean = false;
```

Comment:
```suggestion
  private readonly forked = false;
```

Reply:
Changed it to `private readonly forked: boolean` and assigned the initial value in the constructor.  The `boolean` is needed to ensure a compile time error if we do not definitively set it in the constructor.